bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/OJCloudHandler.java

```diff
@@ -99,64 +101,90 @@ public void dispose() {
+            final ScheduledFuture<?> localUpdateTask = this.updateTask;
+            if (localUpdateTask != null) {
+                localUpdateTask.cancel(false);
+            }
+            this.updateTask = scheduler.schedule(() -> {
+                localUpdateService.updateAllThermostats(getThing().getThings());
+                this.updateTask = null;
+            }, 2, TimeUnit.SECONDS);
+        }
+    }
+
     private void ensureSignIn() {
         if (signInService == null) {
             signInService = new SignInService(configuration, httpClient);
         }
-        final SignInService signInService = this.signInService;
-        if (signInService != null) {
-            signInService.signIn(this::handleSignInDone, this::handleConnectionLost,
+        final SignInService localSignInService = this.signInService;
+        if (localSignInService != null) {
+            localSignInService.signIn(this::handleSignInDone, this::handleConnectionLost,
                     this::handleUnauthorizedWhileSignIn);
         }
     }
 
-    private void handleRefreshDone(@Nullable GroupContentResponseModel groupContentResponse,
+    private void initializationDone(@Nullable GroupContentResponseModel groupContentResponse,
             @Nullable String errorMessage) {
-        logger.trace("OJElectronicsCloudHandler.handleRefreshDone({})", groupContentResponse);
+        logger.trace("OJElectronicsCloudHandler.initializationDone({})", groupContentResponse);
         if (groupContentResponse != null && groupContentResponse.errorCode == 0) {
-            internalRefreshDone(groupContentResponse);
+            internalinitializationDone(groupContentResponse);
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                     (errorMessage == null) ? "Wrong or no result model; Refreshing stoppped" : errorMessage);
-            final RefreshService refreshService = this.refreshService;
-            if (refreshService != null) {
-                refreshService.stop();
+            final RefreshService localRefreshService = this.refreshService;
+            if (localRefreshService != null) {
+                localRefreshService.stop();
             }
         }
     }
 
-    private void internalRefreshDone(GroupContentResponseModel groupContentResponse) {
-        new RefreshGroupContentService(groupContentResponse.groupContents, getThing().getThings()).handle();
-        final OJDiscoveryService discoveryService = this.discoveryService;
-        if (discoveryService != null) {
-            discoveryService.setScanResultForDiscovery(groupContentResponse.groupContents);
+    private void refreshDone(@Nullable SignalRResultModel resultModel, @Nullable String errorMessage) {
+        logger.trace("OJElectronicsCloudHandler.refreshDone({})", resultModel);
+        if (resultModel != null) {
+            new RefreshThermostatsService(resultModel.getThermostats(), resultModel.getThermostatRealTimes(),
```

Comment:
Don't you store it ?

Reply:
What do you mean? What did I miss?

Reply:
Don't you affect the result of this object creation to a variable?

Reply:
No. RefreshThermostatsService only exists for one time use. This service is responsible to analyze the result from the cloud and transmit it to the thermostats. After the job is done, the service has no more date for future use.

Reply:
Ok, it's a bit of strange, not self explaining but fine to me.